### PR TITLE
Fix out of date resource versions of objects in relatedObjects

### PR DESCRIPTION
**Pull request by erkanerol**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2010540

Signed-off-by: Erkan Erol <eerol@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Fix out of date resource versions of objects in relatedObjects
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
Please add release notes with a short bug fix description.

**Comment by nunnatsa** [comment]
/lgtm

 
Member
@erkanerol - please see my comment above about the missing release notes, so I could approve.

**Comment by erkanerol** [comment]
> /lgtm
> 
> @erkanerol - please see my comment above about the missing release notes, so I could approve.

Done. @nunnatsa Did you see my question in the bug? Could you please put a comment there as well?

File: pkg/controller/operands/operand.go [hunk 1]
```diff
@@ -112,6 +112,8 @@ func (h *genericOperand) handleExistingCr(req *common.HcoRequest, key client.Obj
 	}
 
 	if updated {
+		// update resourceVersions of objects in relatedObjects
```
Comment (tiraboschi) [inline pkg/controller/operands/operand.go:115]: https://github.com/openshift/custom-resource-status/blob/master/objectreferences/README.md is not that clear on this,
but I tend to think that this is enough.

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**